cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/bitwarden"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	logLevel       string
+	configPath     string
+	bwUser         string
+	dryRun         bool
+	bwPasswordPath string
+	maxConcurrency int
+
+	config     []bitWardenItem
+	bwPassword string
+}
+
+type bitWardenItem struct {
+	ItemName   string         `json:"item_name"`
+	Field      fieldGenerator `json:"field,omitempty"`
+	Attachment fieldGenerator `json:"attachment,omitempty"`
+	Attribute  fieldGenerator `json:"attribute,omitempty"`
+}
+
+type fieldGenerator struct {
+	Name string `json:"name,omitempty"`
+	Cmd  string `json:"cmd,omitempty"`
+}
+
+const (
+	attributeTypePassword string = "password"
+)
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Whether to actually create the secrets with oc command")
+	fs.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	fs.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	fs.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	fs.StringVar(&o.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	fs.IntVar(&o.maxConcurrency, "concurrency", 1, "Maximum number of concurrent in-flight goroutines to BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
```

Comment:
There is btw `flag.Commandline` for this

Reply:
I did not understand this...i didnt find any commandline in the flag package

Reply:
Its a variable: https://godoc.org/flag#pkg-variables

Its basically just a shortcut to what you do in line 49

Reply:
ok. understood